src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -36,14 +36,11 @@ private static void GetRuntimePropertyInfo(object? value, JsonClassInfo jsonClas
             }
         }
 
-        private static void VerifyValueAndType(object? value, Type? type)
+        private static void VerifyValueAndType(object? value, Type type)
         {
             if (type == null)
             {
-                if (value != null)
-                {
-                    throw new ArgumentNullException(nameof(type));
-                }
+                throw new ArgumentNullException(nameof(type));
```

Comment:
Given this is changing the behavior now for an edge case, please add a test:
Before:
```C#
Console.WriteLine(JsonSerializer.Serialize(null, null)); // outputs "null"
```
Now:
```C#
JsonSerializer.Serialize(null, null); // Throws `ArgumentNullException`.
```

Reply:
The existing `NullObjectOutput` test is failing now because of this change and would need to be updated.
https://github.com/dotnet/runtime/blob/35fe82c2e90fa345d0b6ad243c14da193fb123fd/src/libraries/System.Text.Json/tests/Serialization/SpanTests.cs#L81-L95

Reply:
Yeah I was suspecting there might be a test